### PR TITLE
tests: verify library calculation against generated corpus data using the mainstream implementation

### PR DESCRIPTION
**Pull request by jsign**
This PR adds tests for CommP and piece-size calculations against a corpus of data. 

The test code is pretty straightforward and might not require any explanation. Maybe the interesting part is explaining how this corpus was deterministically generated if, for some reason re-doing or extending the corpus is needed.

The corpus is separated into three pieces:
- Deterministic random data: this data is generated the same way as `jbenet/go-random`, (~`math/random` with a fixed seed)
- All zeros bytes
- 0b11001100 bytes

For each type, we generate data with different sizes between ~2^7 and 2^30 bytes. For each power-of-two range there we select:
- An unimportant size (i.e., some size in the middle point between powers of two) (~normal/most-frequent case).
- Exactly one byte before the fr32 padding threshold.
- Exactly the size which fr32 padding would fit in the current power of two.
- One byte more than fr32 padding will justify not using the consecutive power of two, but the next one.
- Exactly a power of two data size (not entirely interesting, but doesn't hurt).

The interesting part is covering these off-by-one points where fr32 padding leads to different piece sizes. By looking at the corpus data, can be noticed that CommP changes on these off-by-one situations as expected. 

To make this corpus reproducible, I added in `testcase` a script and explanation on how to re-create this data if it wants to be verified or extended in the future.

In short, the script creates this deterministic corpus and calls the Lotus API to calculate CommP and generate our source-of-truth test results.

To be clear: doing `go test ./...` is enough to test the library; re-generating the corpus or running Lotus isn't needed at all.

The corpus is quite big (~450GiB padded pieces) and, in my CPU, takes ~7mins to run the tests. This throughput results because all sub-tests can be run in parallel, so it isn't an indication of a single-calculation throughput. @ribasushi added to this branch a benchmark to measure single calculation throughput.

The test can also be run with the `-short` flag to run a reduced set of cases, which helps to have a faster feedback loop when doing quick changes. (`-short`: <1s vs full: 7min).

### CODE REVIEW

**Comment by ribasushi** [comment]
Excellent stuff, thanks a lot!
